### PR TITLE
Add recipe for Elemental

### LINKED ISSUE #420: Provide Elemental binaries
Now that we have MPICH, should we use the [Elemental from elemental github org](https://github.com/elemental/Elemental), or the [LLNL forked version](https://github.com/LLNL/Elemental)?

### PR DESCRIPTION
**Pull request by christopher-dG**
95% sure this will not work on some platforms but we'll see!

Fix #420

### CODE REVIEW

File: E/Elemental/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+# Collection of sources required to build Elemental
+sources = [
+    GitSource("https://github.com/elemental/Elemental.git",
+              "477e503a7a840cc1a75173552711b980505a0b06"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+if [[ $nbits == 64 ]]; then
```
Comment (giordano) [inline E/Elemental/build_tarballs.jl:16]: On aarch64 our openblas doesn't use LP64, see https://github.com/JuliaPackaging/Yggdrasil/blob/40b71c9995cd0c6c95f20e930d4847c281e804d7/O/OpenBLAS/common.jl#L45
```suggestion
if [[ $nbits == 64 ]] && [[ "${target}" != aarch64-* ]]; then
```

File: E/Elemental/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libEl", :libEl),
+    LibraryProduct("libElSuiteSparse", :libElSuiteSparse),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("METIS_jll"),
+    Dependency("MPICH_jll"),
```
Comment (giordano) [inline E/Elemental/build_tarballs.jl:59]: We don't have `MPICH_jll` for Windows, but we have `MicrosoftMPI_jll`, you can just add it as extra dependency

**Comment by christopher-dG** [comment]
Thanks for the pointers! Looks like Windows builds are still failing with an error I don't know to what to make of: [link](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2745&view=logs&j=64a07938-baf7-57e6-ca68-34433f505c57&t=12c4ac07-1d03-53df-e07d-55312823bafa&l=242)

**Comment by ViralBShah** [comment]
It looks like CMake doesn't know how to find MicrosoftMPI.

**Comment by ViralBShah** [comment]
I wonder if it is thinking of the double colons as a Windows path. Clearly, nobody seems to have built it on Windows before!

**Comment by ViralBShah** [comment]
I guess you've already seen this: https://github.com/elemental/Elemental/issues/21

**Comment by christopher-dG** [comment]
Ah, I hadn't seen that actually. So does this mean we should just exclude Windows platforms for now?

**Comment by giordano** [comment]
I think that's the best idea, yes

**Comment by ViralBShah** [comment]
Merge?

**Comment by ViralBShah** [comment]
I just realized that Elemental is forked by LLNL, and as per Elemental's github page, the LLNL fork is the currently maintained one. It also has a more recent release:

https://github.com/LLNL/Elemental

**Pull request merged**